src/main/java/org/lwes/ArrayEvent.java

```diff
@@ -187,10 +189,10 @@ public void setEventName(String name) {
      */
     @Override
     public void set(String key, FieldType type, Object value) {
-        checkShortStringLength(key, encoding, MAX_FIELD_NAME_SIZE);
+        checkShortStringLength(key, MAX_FIELD_NAME_SIZE);
         if (ENCODING.equals(key)) {
             if (type == FieldType.INT16) {
-                setEncoding((Short) value);
+                setEncoding();
```

Comment:
I am waiting for a second opinion for Nick's comment. Should we throw exception if one tries to set the encoding to anything other than utf-8?